products/compute/helpers/python/instance_start.py

```diff
@@ -0,0 +1,33 @@
+<%
+# Instance On/Off logic
+# This code handles the turning a machine on/off depending on user input.
+# It should be run after all normal create/update/delete logic.
+-%>
+class InstancePower(object):
+    def __init__(self, module, current_status):
+        self.module = module
+        self.current_status = current_status
+        self.desired_status = self.module.params.get('status')
+
+    def run(self):
+        # GcpRequest handles unicode text handling
+        if GcpRequest({'status': self.current_status}) == GcpRequest({'status': self.desired_status}):
+            return
+        elif self.desired_status == 'RUNNING':
+            self.start()
+        elif self.desired_status == 'TERMINATED' or self.desired_status == 'SUSPENDED':
+            self.stop()
```

Comment:
Looks like TERMINATED and SUSPENDED are actually two different states, with the former meaning "powered off" and the latter meaning "hibernating". Do we need a suspend() method that calls https://cloud.google.com/compute/docs/reference/rest/beta/instances/suspend?

Reply:
Does that endpoint work for non-beta? I'm not seeing it in the V1 docs, which makes me think that suspending instances is only a beta thing?

Reply:
I'm inclined to say that we punt on this. If a user wants to suspend an instance, we throw an error that says "we don't currently support this". Users won't be able to suspend instances, but users won't be seeing unexpected changes (module ran, but the instances isn't suspended).

If the suspend API hits v1, we can easily reassess then. If users begin to think that `suspended` always means `terminated`, then changing module behavior down the line will be more frustrating to them.

Reply:
SGTM